File: Darts/src/code/bean/SliderAiSpeed.java
package code.bean;

import javax.swing.JSlider;

public class SliderAiSpeed extends JSlider
{
	private static final int AI_SPEED_MINIMUM = 0; //0s
	private static final int AI_SPEED_MAXIMUM = 2000; //2s
	
	public SliderAiSpeed()
	{
		setMinimum(AI_SPEED_MINIMUM);
		setMaximum(AI_SPEED_MAXIMUM);
		setInverted(true);
		setMajorTickSpacing(100);
	}
}

File: src/main/kotlin/dartzee/bean/SliderAiSpeed.kt
package dartzee.bean

import java.awt.BorderLayout
import javax.swing.JLabel
import javax.swing.JPanel
import javax.swing.JSlider
import javax.swing.border.EmptyBorder
import javax.swing.event.ChangeListener

const val AI_SPEED_MINIMUM = 0 //0s
const val AI_SPEED_MAXIMUM = 2000 //2s

class SliderAiSpeed : JPanel()
{
    private val labelMin = JLabel("Slow")
    private val labelMax = JLabel("Fast")
    private val slider = JSlider()
    var value
        get() = slider.value
        set(value) { slider.value = value }

    init
    {
        layout = BorderLayout()
        add(slider, BorderLayout.CENTER)
        add(labelMin, BorderLayout.WEST)
        add(labelMax, BorderLayout.EAST)

        labelMin.verticalAlignment = JLabel.CENTER
        labelMin.border = EmptyBorder(0, 0, 0, 10)
        labelMax.verticalAlignment = JLabel.CENTER
        labelMax.border = EmptyBorder(0, 10, 0, 0)

        slider.minimum = AI_SPEED_MINIMUM
        slider.maximum = AI_SPEED_MAXIMUM
        slider.inverted = true
    }

    fun addChangeListener(listener: ChangeListener)
    {
        slider.addChangeListener(listener)
    }
}